test/transpiler/transpiler_double_test.ts

```diff
@@ -16,31 +16,814 @@ describe('Transpiler', () => {
+                .transpile('function mul(a) { var x = 1.13; x *= a; return x; }');
+            wrapper.setFunctionName('mul');
+
+            expect(wrapper.call(5)).to.closeTo(5.65, 0.001);
+            expect(wrapper.call(-100)).to.closeTo(-113, 0.001);
+
+            const wrapper2 = new Transpiler()
+                .setSignature('mul', WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64)
+                .transpile('function mul(a) { var x = 3; a *= x; return a; }');
+            wrapper2.setFunctionName('mul');
+
+            expect(wrapper2.call(5.31)).to.closeTo(15.93, 0.001);
+            expect(wrapper2.call(-23.4)).to.closeTo(-70.2, 0.001);
+        });
+
+        it('should handle double shorthand multiplication with array', () => {
+            const wrapper = transpiler
+                .setSignature('mul', WebAssemblyType.FLOAT_64, WebAssemblyType.INT_32_ARRAY)
+                .transpile('function mul(a) { var x = 1.13; x *= a[0]; return x; }');
+            wrapper.setFunctionName('mul');
+
+            expect(wrapper.call([2, 3])).to.closeTo(2.26, 0.001);
+        });
+
+        it('should handle double shorthand division', () => {
+            const wrapper = transpiler
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64)
+                .transpile('function div(a) { var x = 1.13; x /= a; return x; }');
+            wrapper.setFunctionName('div');
+
+            expect(wrapper.call(1.67)).to.closeTo(0.6766467065868263, 0.000000000000001);
+            expect(wrapper.call(-5.15)).to.closeTo(-0.2194174757281553, 0.000000000000001);
+        });
+
+        it('should handle double-int combination shorthand division', () => {
+            const wrapper = transpiler
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.INT_32)
+                .transpile('function div(a) { var x = 1.13; x /= a; return x; }');
+            wrapper.setFunctionName('div');
+
+            expect(wrapper.call(5)).to.closeTo(0.226, 0.0001);
+            expect(wrapper.call(-100)).to.closeTo(-0.0113, 0.00001);
+
+            const wrapper2 = new Transpiler()
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64)
+                .transpile('function div(a) { var x = 3; a /= x; return a; }');
+            wrapper2.setFunctionName('div');
+
+            expect(wrapper2.call(5.31)).to.closeTo(1.77, 0.001);
+            expect(wrapper2.call(-23.4)).to.closeTo(-7.8, 0.001);
+        });
+
+        it('should handle double shorthand division with array', () => {
+            const wrapper = transpiler
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.INT_32_ARRAY)
+                .transpile('function div(a) { var x = 1.13; x /= a[0]; return x; }');
+            wrapper.setFunctionName('div');
+
+            expect(wrapper.call([2, 3])).to.closeTo(0.565, 0.0001);
```

Comment:
I couldn't find any test cases with for loops, while loops and function calls.

Reply:
I'll add tests for function calls. But I don't realy understand what you mean with for and while loop tests. If you mean something like this `while (x < 4.29) { x++; }`, the logic of the double value is already handled in the less then functionality. The while loop itself doesn't care if there is a double value used as it's parameter or not. It just expects a boolean value. Same goes for the for loop.